consensus/istanbul/core/commit.go

```diff
@@ -71,6 +71,9 @@ func (c *core) handleCommit(msg *message, src istanbul.Validator) error {
 	}
 
 	//logger.Error("receive handle commit","num", commit.View.Sequence)
+	if !c.valSet.CheckInSubList(msg.Hash, commit.View, src.Address()) {
+		return errNotFromCommittee
+	}
```

Comment:
How about add this routine in `handleCheckedMsg` to prevent duplicated code?

Reply:
1. The message needs to be decoded in each format to get the `View` used in calculating the sublist. If this routine is moved to `handleCheckedMsg`, the message should be decoded twice. (decoded structures are not an interface)
2.`preprepare` messages don't need to be checked whether in the sublist. If we apply different routines for each message type, `handleCheckedMsg` will be complicated.

Reply:
Why preprepare message does not need to check it?

Reply:
It only has one valid sender(proposer) at a time and is checked with its own routine.